### PR TITLE
fix(server): cloud Public Workspace Logic

### LINKED ISSUE #3957: cloud public workspace logic
currently, we have a field `public` in the workspace schema but it's not good if people only share one page.

So the `public` field will have `PUBLIC`, `ALL`, and `PRIVATE` enums now instead of a boolean value

### PR DESCRIPTION
**Pull request by imf-ali**
fix #3957

### CODE REVIEW

**Comment by imf-ali** [comment]
Hey @himself65 , can you please have a look at this?

**Review by himself65 — CHANGES_REQUESTED** [review]
There's no database migration

File: apps/server/schema.prisma [hunk 1]
```diff
@@ -8,9 +8,15 @@ datasource db {
...
+enum WorkspaceStatus {
+  PUBLIC
+  ALL
+  PRIVATE
+}
+
 model Workspace {
   id             String                    @id @default(uuid()) @db.VarChar
-  public         Boolean
+  public         WorkspaceStatus
```
Comment (himself65) [inline apps/server/schema.prisma]: I think it should be a computed value.

**Review by himself65 — CHANGES_REQUESTED** [review]
rebase the PR to `v0.9.x-staging`

**Comment by imf-ali** [comment]
> There's no database migration

Oh, I will add those

**Comment by himself65** [comment]
Thanks for the contribution! I have rewritten the logic so that we don't need to change the schema

See: https://github.com/toeverything/AFFiNE/pull/3997

**Pull request closed without merging**